### ISSUE TITLE
Doing self.manual_backward(loss) without a training loop

### ISSUE THREAD

**Issue by smartdanny** [PyTorchLightning/pytorch-lightning#9597]
## 🚀 Feature

The ability to do self.manual_backward(loss) without a training loop or trainer associated to the model.

### Motivation

Doing some work involving testing the gradient of the loss with respect to the input image in unsupervised learning. I need to use `manual_backward(`) as to not update the model parameters with the `automatic_optimization`.

### Pitch

Currently, manual_backward() calls `self.trainer.fit_loop.epoch_loop.batch_loop.backward(loss, None, None, *args, **kwargs)`. But if we are not doing this in a training loop, there is no trainer, so we get an `AttributeError` since self.trainer = None. 

### Alternatives

Maybe I can assign the model a dummy trainer? Im open to other hacks since this is totally blocking my research at the moment.

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by carmocca** [comment]
Instead of pursuing `manual_backward` without a `Trainer`, I'd explore finding why the following happens:
> Ive tried freezing the model, but still automatic_optimization seems to unfreeze it and update its parameters when doing loss.backward().

Can you try some dummy example and share it?

Lightning is organized PyTorch and `loss` is a tensor, you shouldn't need to use the Lightning `manual_backward` construct for this
